cassandra-sink/src/test/java/org/springframework/cloud/stream/module/cassandra/sink/CassandraSinkIntegrationTests.java

```diff
@@ -0,0 +1,200 @@
+import org.springframework.cloud.stream.annotation.Bindings;
+import org.springframework.cloud.stream.messaging.Sink;
+import org.springframework.cloud.stream.module.cassandra.test.domain.Book;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.data.cassandra.core.CassandraOperations;
+import org.springframework.data.cassandra.core.CassandraTemplate;
+import org.springframework.integration.support.json.Jackson2JsonObjectMapper;
+import org.springframework.messaging.support.GenericMessage;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import org.apache.cassandra.exceptions.ConfigurationException;
+import org.apache.thrift.transport.TTransportException;
+import org.cassandraunit.utils.EmbeddedCassandraServerHelper;
+
+import com.datastax.driver.core.Cluster;
+import com.datastax.driver.core.querybuilder.QueryBuilder;
+import com.datastax.driver.core.querybuilder.Select;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+
+import reactor.fn.Supplier;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.UUID;
+
+import static org.junit.Assert.assertTrue;
+
+/**
+ * @author Artem Bilan
+ * @author Thomas Risberg
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration(classes = CassandraSinkApplication.class)
+@DirtiesContext
+public abstract class CassandraSinkIntegrationTests {
+
+	public static final String CASSANDRA_KEYSPACE = "cassandraTest";
+
+	private static final String CASSANDRA_STORAGE_CONFIG = "spring-cassandra.yaml";
+
+	private static final int PORT = 9043; // See spring-cassandra.yaml - native_transport_port
+
+    @Autowired
+    @Bindings(CassandraSink.class)
+    protected Sink sink;
+
+    @Autowired
+    ConfigurableApplicationContext applicationContext;
+
+	private static Cluster cluster;
+
+	private static CassandraOperations cassandraTemplate;
+
+	@BeforeClass
+	public static void setUp() throws ConfigurationException, IOException, TTransportException {
+		EmbeddedCassandraServerHelper.startEmbeddedCassandra(CASSANDRA_STORAGE_CONFIG, "target/embeddedCassandra");
+		cluster = Cluster.builder()
+				.addContactPoint("localhost")
+				.withPort(PORT)
+				.build();
+
+		cluster.connect().execute(String.format("CREATE KEYSPACE IF NOT EXISTS %s" +
+				"  WITH REPLICATION = { 'class' : 'SimpleStrategy', 'replication_factor' : 1 };", CASSANDRA_KEYSPACE));
+
+		cassandraTemplate = new CassandraTemplate(cluster.connect(CASSANDRA_KEYSPACE));
+
+	}
+
+	@AfterClass
+	public static void cleanup() {
+		if (cluster != null) {
+			cluster.close();
+		}
+		EmbeddedCassandraServerHelper.cleanEmbeddedCassandra();
+	}
+
+	@WebIntegrationTest({"spring.cassandra.port=9043", 
+		"spring.cassandra.keyspace=cassandraTest",
```

Comment:
switching to use constants. For dynamic ports, not even sure that is possible with Cassandra, There are quite a few ports required in the yml file, not sure if we could set system properties at  runtime to override them. I did change a few unneeded ones to use '0'.